### PR TITLE
Dev/212-210-209: Testcase Deletion and scrolling on new testcase / Problem Editor Integration

### PR DESCRIPTION
**Pull request by Johnny850807**
resolve: #212 
resolve: #210 
resolve: #209 

Dev/212-210-209: Testcase Deletion and scrolling on new testcase / Problem Editor Integration

### CODE REVIEW

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: src/app/problem/editor/context.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        case ACTION_UPDATE_COMPILATION:
+            return newProblem({
+                ...problem,
+                languageEnvs: updateLanguageEnv(problem.languageEnvs, action.languageEnv)
+            });
+        case ACTION_UPDATE_LANGUAGE_ENV:
+            return newProblem({
+                ...problem,
+                languageEnvs: updateLanguageEnv(problem.languageEnvs, action.languageEnv)
+            });
```
Comment (Wally5077) [inline src/app/problem/editor/context.js:43]: NIT: It seems like there is the same updating logic, so should we combine these two cases?

File: src/app/problem/editor/right/testcases/usecase.js [hunk 2]
```diff
@@ -4,40 +4,51 @@ import {TestcaseIOsPatch} from "../../../../../services/ProblemService";
...
-
-                editing: true // new testcase --> editing it immediately
-            })]);
+        const nextId = testcaseEdits.length === 0 ? 0 : generateNextTestcaseId(testcaseEdits);
+        const newTestcaseEdit = new TestcaseEdit({
+            id: nextId.toString(),
+            name: nextId.toString(),
+            timeLimit: 1,
+            memoryLimit: 1,
+            outputLimit: 1,
```
Comment (Chaoyuuu) [inline src/app/problem/editor/right/testcases/usecase.js:18]: In the current Judge Girl, it has `timeLimit: 1.0s` and `memoryLimit: 64MB` by default.
Please update these two limitations to the correct default value.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review Done

File: src/app/problem/editor/right/Visible.js [hunk 3]
```diff
@@ -1,31 +1,37 @@
...
+
     const onVisibleButtonClick = (isVisibleButtonClicked) => {
-        setIsOnClicked(isVisibleButtonClicked);
-        problemService.updateProblemVisible(problemId, isVisibleButtonClicked)
-            .then(() => {
-                console.log("The problem's visibility has been modified");
-                markProblemsDirty();
-            });
+        if (isVisibleButtonClicked !== isOnClicked) {
+            setIsOnClicked(isVisibleButtonClicked);
```
Comment (Chaoyuuu) [inline src/app/problem/editor/right/Visible.js:21]: `isVisibleButtonClicked` and `isOnClicked` should be more readable.
Please rename them.

**Review by Chaoyuuu — CHANGES_REQUESTED** [review]
Review Done

**Pull request merged**